src/main/java/com/microsoft/azure/spring/data/documentdb/DocumentDbFactory.java

```diff
@@ -9,40 +9,53 @@
 import com.microsoft.azure.documentdb.ConnectionPolicy;
 import com.microsoft.azure.documentdb.ConsistencyLevel;
 import com.microsoft.azure.documentdb.DocumentClient;
-import com.microsoft.azure.spring.data.documentdb.common.GetHashMac;
-import com.microsoft.azure.spring.data.documentdb.common.PropertyLoader;
+import com.microsoft.azure.spring.data.documentdb.common.TelemetryProperties;
+import com.microsoft.azure.spring.data.documentdb.common.TelemetryProxy;
 import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
 
-public class DocumentDbFactory {
+import java.util.HashMap;
+import java.util.Map;
 
-    private static final String USER_AGENT_SUFFIX = Constants.USER_AGENT_SUFFIX + PropertyLoader.getProjectVersion();
+public class DocumentDbFactory {
 
     private DocumentClient documentClient;
+    private final TelemetryProxy telemetryProxy;
 
     public DocumentDbFactory(String host, String key) {
-        this(host, key, true);
-    }
-
-    public DocumentDbFactory(String host, String key, boolean isBiEnabled) {
         Assert.hasText(host, "host must not be empty!");
         Assert.hasText(key, "key must not be empty!");
 
         final ConnectionPolicy policy = ConnectionPolicy.GetDefault();
 
-        String userAgent = ";" + USER_AGENT_SUFFIX;
-        if (isBiEnabled && GetHashMac.getHashMac() != null) {
-            userAgent += ";" + GetHashMac.getHashMac();
-        }
-        policy.setUserAgentSuffix(userAgent);
```

Comment:
please keep old ua. old ua is setting at service side, it provides telemetry which is not able to be collected by spring boot starters, like # of subs, maybe subs distribution per location, resource size etc. telemetry collected in ai is just a supplement.

Reply:
OK. I remove that as it seems duplicated. Will recover that.